### ISSUE TITLE
Provide an ISO with latest stable kernel for Ryzen APU owners

### ISSUE THREAD

**Issue by JackMont** [QubesOS/qubes-issues#4919]
There are many people complaining about Ryzen issues (especially the 2200G and 2400G with integrated graphics), which apparently gets solved with kernel 4.18+
Please just release a new ISO for these people already, so all those related issues will be gone. I happen to be one of them and I just spent an entire day without any working installation in the end.

**Comment by marmarek** [comment]
Once [4.19 lands in stable](https://github.com/QubesOS/updates-status/issues/949), releasing 4.0.2 ISO may be a good idea.
On the other hand, the process takes some time, so it may be a good idea to wait for Fedora 30 final release (about a month from now).

**Comment by JackMont** [comment]
Is it possible you could provide us folks (and the future folks) with 2200G and 2400G CPUs with a quick inofficial fix? It would be nice not having to wait that long. Personally I don't have access to a variety of other hardware and I fail at building my own ISO...

**Comment by andrewdavidwong** [comment]
> Once [4.19 lands in stable](https://github.com/QubesOS/updates-status/issues/949), releasing 4.0.2 ISO may be a good idea.
> On the other hand, the process takes some time, so it may be a good idea to wait for Fedora 30 final release (about a month from now).

Is this a "won't do," then?

**Comment by 0spinboson** [comment]
> Is it possible you could provide us folks (and the future folks) with 2200G and 2400G CPUs with a quick inofficial fix? It would be nice not having to wait that long. Personally I don't have access to a variety of other hardware and I fail at building my own ISO...

disabling modesetting doesn't help, in allowing you to get to a command line so you can run the updater after installation?

**Comment by JackMont** [comment]
> 
> 
> > Is it possible you could provide us folks (and the future folks) with 2200G and 2400G CPUs with a quick inofficial fix? It would be nice not having to wait that long. Personally I don't have access to a variety of other hardware and I fail at building my own ISO...
> 
> disabling modesetting doesn't help, in allowing you to get to a command line so you can run the updater after installation?

I tried everything. I always end up with X startup failed, aborting installation followed by a Pane is dead (yes I tried everything suggested in that old thread). No matter if I add nomodeset mapbs=1 noexitboot=1 and no matter the UEFI BIOS settings (I enabled CPU virtualization and IOMMU).

It would be just way easierto have 4.18 kernel and just easily go through the installation without hiccups and without needing to spend days trying to figure out & solve the problem.

**Comment by fepitre** [comment]
HI,

Up to what would be officially released or not, for several reasons I
have a mirror with ISO built having both kernel and kernel-latest
packages (version in qubes-linux-kernel master). Currently it's
restricted but if there is a need, I can manage to open the mirror.

Please note that would be unofficial/unsupported ISO. ISO are
bi-weekly/monthly updated.

Best,

Frédéric

**Comment by JackMont** [comment]
@fepitre That would be very nice. It needs to run 4.18 or higher. Could you provide the link in here to that version? I'd consider this issue solved then. People with a similar problem will find this issue and it'll be solved for them as well.

**Comment by fepitre** [comment]
Currently it's restricted access. I will try to manage to open it this
week-end.

Best,

**Comment by JackMont** [comment]
@fepitre alternatively I'd also be very happy with just a mediafire, dropbox, google cloud, torrent link or whatever your choice of preference is, if it makes this easier & faster.

**Comment by fepitre** [comment]
Please wait this week-end. Currently, the ISOs are custom for specific
needs. It's just adding a rule in my build pipeline for generic qubes
iso and adjusting the access.

Best,

**Comment by JackMont** [comment]
Thank you very much. Looking forward to finally being able to install Qubes OS properly & without a hassle.

**Comment by JackMont** [comment]
@fepitre Did you succeed in doing it?

**Comment by fepitre** [comment]
@JackMont: you will find a latest build on my web server: http://mirror.notset.fr/qubes/iso/ (https is arriving soon...). It is signed with my bot key (1C87 14D6 40F3 0457 EC95  3050 6569 46BA 873D DEC1). I will push soon a separate repository with details of the kickstart used instead of the one, key information, CI pipeline etc. The iso has been generated with the kickstart that you can find into the repo too. Globally, I added all the official Qubes repositories, no package has been built from my side, excluded kernel component and replaced it by kernel-latest. If you need any supplementary information, I will provide it.

**Comment by fepitre** [comment]
Please note that I'm doing some maintenance on my servers, I have limited the bandwidth to 1M/s which will be set higher in the afternoon (sorry intersection of several phenomena).

**Comment by JackMont** [comment]
@fepitre Thanks a ton! Will try an install as soon as it is downloaded and let you know how it went.

**Comment by fepitre** [comment]
Sure thank you. Do not hesitate to download it with something like 'uGet' in case I restart my web server in the afternoon.

**Comment by JackMont** [comment]
@fepitre Thanks for the iso. Installation went very smooth without the need of any adjustments.
There's a problem with attaching USBs though. I slide the drive in, press the attach button to an AppVM and it just doesn't do anything. Updated & upgraded everything, rebooted and tried on many different AppVMs (both Debian and Fedora 29 based). USB drives just won't attach.

**Comment by fepitre** [comment]
So at first it's great you succeeded to install Qubes! For sys-usb, have you enabled the sys-usb VM?

**Comment by JackMont** [comment]
@fepitre I didn't change anything and it just suddenly started working. Is it normal that upon attaching the drive to an AppVM, that one has to additionally mount the drive via /dev/x onto a folder? I thought it would auto-mount.

All in all my issue is therefore solved and I can successfully use all features of Qubes OS, love it so far!

Minor issues I've noticed:
1. I did all updates after the install but for whatever reason the Update Launcher and Qubes Manager still show an update notification for the Debian 9 template-vm. It doesn't go away after clicking the update button and rebooting several times.
2. Time is off in dom0, for some Qubes it's different than others. Is this on purpose?
3. The DispVM of anon-whonix (AppVM) doesn't get its browser updated when the update is made in anon-whonix. -> it would be great if there was a one-click button to generate a new DispVM which is up to date with its template

**Comment by fepitre** [comment]
Great if it works. Probably a reboot was needed to properly initialize the pci devices. I also encounter this problem of mounting manually when I attach a partition instead of the block itself (never had time to dig it into more). Normally, it appears in the 'File' as any other linux distro. For other issues, have a look on existing issues or don't hesitate to open one if it's suitable. Have fun and welcome on Qubes!

**Comment by fepitre** [comment]
I think we can close this issue as I built latest ISO almost periodically and the ISO are accessible on my private mirror http://mirror.notset.fr/qubes/iso. Please feel free to reopen this issue if it was targeting another work to be done.
